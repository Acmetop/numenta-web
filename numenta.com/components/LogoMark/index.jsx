// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import React from 'react'

import Image from '../Image'
import LogoMarkDark from './images/numenta-mark-dark.svg'
import LogoMarkLite from './images/numenta-mark-lite.svg'

import styles from './index.css'


/**
 *
 */
const LogoMark = ({color}) => {
  const Marks = {
    dark: LogoMarkDark,
    lite: LogoMarkLite,
  }
  const Logo = Marks[color]

  return (
    <Image
      alt="Numenta Logo Mark"
      className={styles.mark}
      shadow={false}
      src={Logo}
      title="Numenta"
    />
  )
}

LogoMark.propTypes = {
  color: React.PropTypes.oneOf(['dark', 'lite']).isRequired,
}

LogoMark.defaultProps = {
  color: 'dark',
}

export default LogoMark
