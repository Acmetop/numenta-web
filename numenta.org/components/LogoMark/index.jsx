// Numenta shared web components
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import Image from '../../../components/Image'

import LogoMarkDark from './images/numenta-mark-dark.svg'
import LogoMarkLite from './images/numenta-mark-lite.svg'
import styles from './index.css'


/**
 * Logo Mark - React view component.
 */
const LogoMark = ({color}) => {
  const Marks = {
    dark: LogoMarkDark,
    lite: LogoMarkLite,
  }
  const Logo = Marks[color]

  return (
    <Image
      alt="Numenta Logo Mark"
      className={styles.mark}
      shadow={false}
      src={Logo}
      title="Numenta"
    />
  )
}

LogoMark.propTypes = {
  color: React.PropTypes.oneOf(['dark', 'lite']).isRequired,
}

LogoMark.defaultProps = {
  color: 'dark',
}

export default LogoMark
