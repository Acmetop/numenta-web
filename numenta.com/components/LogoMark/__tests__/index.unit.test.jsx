import React from 'react'
import renderer from 'react-test-renderer'

import LogoMark from '../../LogoMark'


describe('<LogoMark /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <LogoMark />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
