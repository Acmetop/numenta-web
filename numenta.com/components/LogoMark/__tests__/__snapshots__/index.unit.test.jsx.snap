exports[`LogoMark React component Renders Dark correctly 1`] = `
<img
  alt="Numenta Logo Mark"
  className="image"
  onClick={undefined}
  src="filename.ext"
  title="Numenta" />
`;

exports[`LogoMark React component Renders Default correctly 1`] = `
<img
  alt="Numenta Logo Mark"
  className="image"
  onClick={undefined}
  src="filename.ext"
  title="Numenta" />
`;

exports[`LogoMark React component Renders Lite correctly 1`] = `
<img
  alt="Numenta Logo Mark"
  className="image"
  onClick={undefined}
  src="filename.ext"
  title="Numenta" />
`;
